### PR TITLE
Implemented FreeEagle defense

### LINKED ISSUE #74: [FEATURE]  Integrate FreeEagle as a Data-Free Defense
## Summary

Integrate the FreeEagle backdoor detection method into Mithridatium as a new data-free defense option. FreeEagle detects complex neural trojans without requiring access to clean training data by analyzing neuron behavior and anomaly patterns.

This issue focuses on creating a working integration that allows FreeEagle to run on a given model within our pipeline.

## Acceptance Criteria
- [ ] FreeEagle is runnable from within the Mithridatium repository.

- [ ] A user can pass a model path and dataset name as input.

- [ ] The defense produces:
	•	An anomaly score
	•	A detection verdict (clean/backdoored)
	•	Generated inspection artifacts (if applicable)

- [ ] Output is integrated into Mithridatium’s structured reporting format (JSON + CLI output).

- [ ] FreeEagle runs successfully on at least:
	•	One clean model
	•	One backdoored model

## Tasks that need to be completed for this feature

	•	Clone or fork the FreeEagle repository.

	•	Review paper + implementation to understand:
	       •	Required inputs
	       •	Dataset assumptions
	       •	Model format expectations

	•	Identify minimal components needed for inspection:
	       •	backdoor_inspection_new.py
	       •	anomaly metric computation logic

	•	Create a wrapper module: mithridatium/defenses/freeeagle.py

	•	Implement:
	       •	Model loading adapter
	       •	Configuration mapping (dataset name, num classes, etc.)
	       •	Execution wrapper

	•	Convert FreeEagle outputs into Mithridatium JSON report format.

	•	Add CLI option: mithridatium scan --defense freeeagle --model path/to/model.pth

	•	Test on:
	       •	Clean model
	       •	Known backdoored model

## Additional Context

FreeEagle is particularly valuable because:
	•	It is data-free
	•	It does not require clean reference data
	•	It works in settings where other defenses fail

This aligns well with Mithridatium’s goal of verifying downloaded pretrained models.

### PR DESCRIPTION
**Pull request by GustavoLucca**
## Description

Fixes #74
FreeEagle implementation has begun by setting up a skeleton, a wrapper, and wiring it to the CLI.
Update: FreeEagle has been fully implemented and tested across different types of backdoors.

# 1. What was Changed

Began the implementation of FreeEagle by creating files containing the logic, the wrapper, and the CLI. 
Update: Fully wired arguments into the CLI, ran various tests, and confirmed the defense works across models.

# 2. Why it was Changed

FreeEagle is a white-box data free backdoor detection defense, so it will be useful for users who have access to the internal model weights but not the data, since they can still test it and see if it is compromised or not.

# 3. How it was Changed

A file structure was created under mithridatium/defenses for FreeEagle, and tests were created to ensure the proper functionality of the scripts.
Update: Prepared the FreeEagle report structure, Finalized FreeEagle CLI integration with all parameters, and tested our defense implementation across the clean and all the different types of backdoored models.

### CODE REVIEW

**Review by pelumitegbe — APPROVED** [review]
Nice work on this PR. This is a substantial addition to the project.

I like that FreeEagle was integrated in a way that keeps the core algorithm separate from the Mithridatium-facing wrapper and CLI/reporting pieces. That makes the implementation easier to reason about and should help future maintenance as we compare or tune defenses.

I also appreciate that this PR includes tests and documentation, not just the defense logic itself. That makes it much easier for the rest of the team to use and build on.

This looks good to merge

File: mithridatium/defenses/FreeEagle/backdoor_inspection_new.py [hunk 1]
```diff
@@ -0,0 +1,493 @@
+# detecting malicious samples that triggers backdoor via:
```
[the remark this answers is not in the document]
Comment (GustavoLucca) [reply mithridatium/defenses/FreeEagle/backdoor_inspection_new.py:1]: Yes, also to give context to the user of how we implemented and changed the defense for our pipeline.

[the remark this answers is not in the document]
Comment (pelumitegbe) [reply mithridatium/defenses/FreeEagle/backdoor_inspection_new.py:1]: I see, great work
